cura/Machines/Models/MachineListModel.py

```diff
@@ -25,10 +25,13 @@ class MachineListModel(ListModel):
     IsOnlineRole = Qt.ItemDataRole.UserRole + 5
     MachineCountRole = Qt.ItemDataRole.UserRole + 6
     IsAbstractMachine = Qt.ItemDataRole.UserRole + 7
+    ListTypeRole = Qt.ItemDataRole.UserRole + 8
```

Comment:
Not sure if this is the right name for it. All the other ones can be applied to the machine, not whether it's a (non-printer) button or not. (Speaking of which, I now just noticed that IsAbstractMachine doesn't have 'Role' appended to it's name.)

I'm OK with keeping it this way if we can't come up with a better name.

Reply:
I agree, the name is not perfect. I changed it to ComponentType, I think this makes it a bit clearer this variable is used to decide which component to use with the model. 

Not much better, but I was getting a bit of analysis paralysis trying to name this so I went with the best I could come up with.